katana/native/projects/static/doProjectHandling.js

```diff
@@ -0,0 +1,735 @@
+//
+//
+/*
+/// -------------------------------------------------------------------------------
+
+Project File Data Handler 
+
+Author: 
+Date: 
+
+The functions in this module are designed specifically for handling project XML files
+for the warrior framework. 
+
+It is expected to work with the editProject.html file and the calls to editProject in 
+the views.py python for Django. 
+/// -------------------------------------------------------------------------------
+
+*/
+if (typeof jsonAllProjectPages === 'undefined') {
+ jsonAllProjectPages = { };
+} else {
+	//alert("Already there...");
+}
+var jsonProjectObject = []; 
+var jsonTestSuites = [];			// for all Suites
+
+
+/// -------------------------------------------------------------------------------
+// 
+/// -------------------------------------------------------------------------------
+function getRandomID() {
+  min = Math.ceil(1);
+  max = Math.floor(2000);
+  return Math.floor(Math.random() * (max - min)) + min;
+  
+}
+
+function jsUcfirst(string) 
+{
+    return string.charAt(0).toUpperCase() + string.slice(1);
+}
+/// -------------------------------------------------------------------------------
+// Sets up the global project data holder for the UI. 
+// This is called from the correspoding HTML file onLoad event 
+// or when a new XML file is loaded into the interface.
+// 
+// Two variables are set when this function is called; 
+// 1. jsonProjectObject 
+// 2. jsonTestSuites is set to point to the Testsuites data structure in
+//    the jsonProjectObject
+//
+/// -------------------------------------------------------------------------------
+function mapFullProjectJson(myobjectID){
+	var sdata = katana.$activeTab.find("#listOfTestSuitesForProject").text();
+	katana.$activeTab.find("#listOfTestSuitesForProject").hide();
+	katana.$activeTab.find('#savefilepath').hide();  // To remove later...
+
+	var jdata = sdata.replace(/'/g, '"');
+	console.log('Mapping data ... ' + typeof(sdata) + ' is [' + sdata + "] " + sdata.length);  // This jdata is a string ....
+	//console.log(jdata);                  // I show it as such. 
+	jsonAllProjectPages[myobjectID] = JSON.parse(sdata); 
+	//console.log(typeof(jsonAllProjectPages[myobjectID]));
+	jsonProjectObject =  jsonAllProjectPages[myobjectID]; 
+	jsonTestSuites = jsonProjectObject['Testsuites']; 
+	mapProjectJsonToUi(jsonTestSuites);  // This is where the table and edit form is created. 
+	fillProjectDefaultGoto();
+	console.log("Adding defaults ");
+	katana.$activeTab.find('#default_onError').on('change',fillProjectDefaultGoto );
```

Comment:
instead of using onchange here
can apply katana-change directly on the object as an attribute